File: app/src/main/java/com/android/retrofitsampleapp2/iu/projects/GitProjectAdapter.java
package com.android.retrofitsampleapp2.iu.projects;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.android.retrofitsampleapp2.R;
import com.android.retrofitsampleapp2.domain.GitProjectEntity;

import java.util.List;

public class GitProjectAdapter extends RecyclerView.Adapter<GitProjectViewHolder> {

    private List<GitProjectEntity> data;//Завели приватный список (закэшировали)

    public void setData(List<GitProjectEntity> project) {
        data = project;
        notifyDataSetChanged(); //для того чтобы у всех обнавлялось
    }

    // создаем сам ViewHolder
    @NonNull
    @Override
    public GitProjectViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new GitProjectViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_git_project, parent, false));
    }

    // кладем данные во ViewHolder
    @Override
    public void onBindViewHolder(@NonNull GitProjectViewHolder holder, int position) {
        holder.bind(getItem(position));
    }

    private GitProjectEntity getItem(int pos) {
        return data.get(pos);
    }

    @Override
    public int getItemCount() {
        return data.size();
    }
}

File: app/src/main/java/com/android/retrofitsampleapp2/iu/projects/GitProjectAdapter.kt
package com.android.retrofitsampleapp2.iu.projects

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.android.retrofitsampleapp2.R
import com.android.retrofitsampleapp2.domain.GitProjectEntity

class GitProjectAdapter(
    private var data: List<GitProjectEntity>
) : RecyclerView.Adapter<GitProjectViewHolder>() {

    fun setData(project: List<GitProjectEntity>) {
        data = project
        notifyDataSetChanged() //для того чтобы у всех обнавлялось
    }

    // создаем сам ViewHolder
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): GitProjectViewHolder {
        return GitProjectViewHolder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.item_git_project, parent, false)
        )
    }

    // кладем данные во ViewHolder
    override fun onBindViewHolder(holder: GitProjectViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    private fun getItem(pos: Int): GitProjectEntity = data[pos]

    override fun getItemCount(): Int = data.size

}